evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -116,6 +117,16 @@ def _add_training_data_to_X_Y(self, X, y, X_train, y_train):
                 self.gap,
             )
 
+            # Properly fill in the dates in the gap
```

Comment:
This was a pre-existing bug where we fill in the gap between the training and testing data with the last row of training data, but that means creating an irregularly spaced time index.